scalableminds/webknossos
# Check datastore health when requests error

This PR adds a very simple health check mechanism to our request module. When a request failed (for whatever reason), we will ping the datastore health endpoint and display a warning in case the status is not 200.

@fm3 The healthcheck is currently done by pinging `data/health` on wk (which redirects to the actual datastore). I think that's alright, since it doesn't really matter whether the datastore is up when wk is down. Do you agree?

### Mailable description of changes:
 - WebKnossos will notify users when the datastore is not available (e.g., due to an update) and ask them 
to check back later.

### Steps to test:
- ~This was quite hard to test, since the datastore and wk itself are run together locally. I disabled the client-validation for bulk task creation and changed the `catch` in `checkDataStoreHealth` to `then`. As long as the health endpoint works as expected, everything should work fine. But maybe @fm3 knows of a better way to test this.~
- Set up: [Here is how to set up a standalone-datastore locally](https://github.com/scalableminds/webknossos-wiki/wiki/Set-up-a-standalone-datastore-locally)
- Create a new task & try to get a new task --> should work
- Get another new task -> should fail with no datastore-related error message
- Open a dataset --> should work
- Kill the datastore
- Create a new task & try to get a new task --> should fail with a warning about the datastore
- Open a dataset --> shouldn't display data (but instead warn about the datastore)

### Issues:
- contributes to #2604

------
